drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -689,6 +689,30 @@ static void vc4_hdmi_set_spd_infoframe(struct drm_encoder *encoder)
 	vc4_hdmi_write_infoframe(encoder, &frame);
 }
 
+static void vc4_hdmi_set_hdmi_infoframe(struct drm_encoder *encoder)
+{
+	struct vc4_hdmi *vc4_hdmi = encoder_to_vc4_hdmi(encoder);
+	struct drm_connector *connector = &vc4_hdmi->connector;
+	struct drm_crtc *crtc = encoder->crtc;
+	const struct drm_display_mode *mode = &crtc->state->adjusted_mode;
+	union hdmi_infoframe frame;
+	int ret;
+
+	ret = drm_hdmi_vendor_infoframe_from_display_mode(&frame.vendor.hdmi,
+							  connector,
+							  mode);
+	if (ret < 0) {
+		DRM_ERROR("couldn't fill HDMI infoframe (%d)\n", ret);
```

Comment:
https://elixir.bootlin.com/linux/latest/source/drivers/gpu/drm/drm_edid.c#L5809
>   Note that there's is a need to send HDMI vendor infoframes only when using a
  4k or stereoscopic 3D mode. So when giving any other mode as input this
  function will return -EINVAL, error that can be safely ignored.

Logging an error is therefore a bad move.